modin/numpy/arr.py

```diff
@@ -1469,6 +1559,247 @@ def sum(
+        dtype=None,
+        subok=True,
+    ):
+        check_kwargs(where=where, casting=casting, order=order, subok=subok)
+        result = self._query_compiler._isinf()
+        return fix_dtypes_and_determine_return(result, self._ndim, dtype, out, where)
+
+    def _isnan(
+        self,
+        /,
+        out=None,
+        *,
+        where=True,
+        casting="same_kind",
+        order="K",
+        dtype=None,
+        subok=True,
+    ):
+        check_kwargs(where=where, casting=casting, order=order, subok=subok)
+        result = self._query_compiler.isna()
+        return fix_dtypes_and_determine_return(result, self._ndim, dtype, out, where)
+
+    def _isnat(
+        self,
+        /,
+        out=None,
+        *,
+        where=True,
+        casting="same_kind",
+        order="K",
+        dtype=None,
+        subok=True,
+    ):
+        check_kwargs(where=where, casting=casting, order=order, subok=subok)
+        result = self._query_compiler._isnat()
+        return fix_dtypes_and_determine_return(result, self._ndim, dtype, out, where)
+
+    def _isneginf(self, out=None):
+        result = self._query_compiler._isneginf()
+        return fix_dtypes_and_determine_return(result, self._ndim, out=out)
+
+    def _isposinf(self, out=None):
+        result = self._query_compiler._isposinf()
+        return fix_dtypes_and_determine_return(result, self._ndim, out=out)
+
+    def _iscomplex(self):
+        result = self._query_compiler._iscomplex()
+        return fix_dtypes_and_determine_return(result, self._ndim)
+
+    def _isreal(self):
+        result = self._query_compiler._isreal()
+        return fix_dtypes_and_determine_return(result, self._ndim)
+
+    def _logical_not(
+        self,
+        /,
+        out=None,
+        *,
+        where=True,
+        casting="same_kind",
+        order="K",
+        dtype=None,
+        subok=True,
+    ):
+        check_kwargs(where=where, casting=casting, order=order, subok=subok)
+        result = self._query_compiler._logical_not()
+        return fix_dtypes_and_determine_return(result, self._ndim, dtype, out, where)
+
+    def _logical_binop(
+        self, qc_method_name, x2, out, where, casting, order, dtype, subok
+    ):
+        check_kwargs(where=where, casting=casting, order=order, subok=subok)
+        if is_scalar(x2):
+            return fix_dtypes_and_determine_return(
+                getattr(self, qc_method_name)(x2),
+                self._ndim,
+                dtype,
+                out,
+                where,
+            )
+        caller, callee, new_ndim, kwargs = self._binary_op(x2)
+        # breakpoint()
+        if caller._query_compiler != self._query_compiler:
+            # In this case, we are doing an operation that looks like this 1D_object.op(2D_object).
+            # For Modin to broadcast directly, we have to swap it so that the operation is actually
+            # 2D_object.op(1D_object).
+            caller, callee = callee, caller
```

Comment:
Why are we re-swapping the results of `_binary_op` here?

Reply:
This is a mistake, I've removed it.